src/hooks/use-text-filter.js

```diff
@@ -10,10 +12,57 @@ const DEBOUNCE_TIME = 200;
  * postprocessing of data from Realm
  * @param {*} query the text filter query to use for searching
  */
-function useTextFilter(query) {
+function useTextFilter(query, activeContentTab) {
     const [filterEvent, setFilterEvent] = useState(null);
     const [results, setResults] = useState(null);
 
+    /**
+     * Gets all results
+     * @param {Array} filterResults
+     * @returns {Array}
+     */
+    const getAll = filterResults => {
+        return filterResults.map(r => {
+            switch (r.mediaType) {
+                case 'youtube':
+                case 'twitch':
+                    return {
+                        ...new SearchVideoResult(r),
+                    };
+                case 'podcast':
+                    return {
+                        ...new SearchPodcastResult(r),
+                    };
+                default:
+                    return {
+                        ...new SearchArticleResult(r),
+                    };
+            }
+        });
+    };
+
+    /**
+     * Gets the filterResults from a mediaType
+     * @param {Array} filterResults
+     * @param {string} mediaType
+     * @param {*} ObjectClass
+     * @returns {Array}
+     */
+    const getResultsFromType = (filterResults, mediaType, ObjectClass) => {
+        return filterResults.reduce((results, r) => {
+            if (r.mediaType === mediaType) {
+                results.push({ ...new ObjectClass(r) });
+            } else if (
+                mediaType === 'video' &&
+                (r.mediaType === 'twitch' || r.mediaType === 'youtube')
+            ) {
+                results.push({ ...new ObjectClass(r) });
+            }
+
+            return results;
```

Comment:
Help me remember, did we think of reverting back to actual classes. I'm afraid this will give run time errors later, if I remove/rename the class at some point.

Reply:
Lol I don't remember either. I can revert back

Reply:
yup.Go ahead and revert.